src/App.tsx

```diff
@@ -1,24 +1,125 @@
-import React, { useState } from 'react';
+import React, { useReducer } from 'react';
 import { StyleSheet, Text, View } from 'react-native';
 import AddItem from './components/AddItem';
 import AddPartyMember from './components/AddPartyMember';
 import Tax from './components/Tax';
 import Tip from './components/Tip';
 
+enum Actions {
+  ADD_ITEM = 'add_item',
+  ADD_PARTY_MEMBER = 'add_party_member',
+  SET_TIP = 'set_tip',
+  SET_TAX = 'set_tax',
+}
+
+type AppStateType = {
+  total: number;
+  items: {
+    [name: string]: {
+      price: number;
+      purchaserList: string[];
+    };
+  };
+  party: string[];
+  tip: number;
+  tax: number;
+};
+
+type ActionArgs = {
+  type: Actions;
+
+  // item name or party member name
+  name: string;
+
+  // item details
+  price: number;
+  purchaserList: string[];
+
+  // tax or tip amount
+  amount: number;
+};
+
 export default function App() {
-  const [total, setTotal] = useState(0);
-  const [items, setItems] = useState({});
-  const [party, setParty] = useState([]);
-  const [tip, setTip] = useState(0);
-  const [tax, setTax] = useState(0);
-
-  const addItem = (itemName: string, itemPrice: number, itemPurchaserList: Array<string>) => {
-    setItems({ ...items, [itemName]: [itemPrice, itemPurchaserList] });
-    setTotal(Number((total + itemPrice).toFixed(2)));
+  const initialState: AppStateType = {
+    total: 0,
+    items: {},
+    party: [],
+    tip: 0,
+    tax: 0,
+  };
+
+  const reducer = (state: AppStateType, action: Partial<ActionArgs>) => {
+    switch (action.type) {
+      case Actions.ADD_ITEM: {
+        const updatedItems = {
+          ...state.items,
+          [action.name]: {
```

Comment:
open to other thoughts about this. I changed the item to be an object of the item price and purchaser list rather than just `[price, purchaserList]`. I'm not sure if storing the details in a list would make it extendable for future use if we were to add more details (idk what they would be lol), and it's a list of mixed types of a number and a list of purchaser names.

Reply:
I don't think the original item was a list of object; looks like it was a key-value store:
```
setItems({ ...items, [itemName]: [itemPrice, itemPurchaserList] });
```
where key is itemName and value is a list? It does sound similiar to what you're suggesting. I think we chose for the key value structure so updating the purchaser list when adding people to associate will be a quicker look up. Would your change affect this?

Reply:
^ Actually, looking at your change again, I think we're maintaining the same structure for the items state. No objections here. 😃

Reply:
Oh what I meant was that each item previously was mapped to a list containing the price and purchaser list. The change in this PR is that each item is now mapped to another object containing the price and purchaser list to separate the types and data storage of those details.

Reply:
Oh gotcha, so instead of each item looking like:
```
[itemPrice: number, itemPurchaseList: string[]]
```
it now looks like: 
```
{
      price: number;
      purchaserList: string[];
    }
```
That makes sense to me. 👍